### PR TITLE
Add type definitions for StyleOptions

### LINKED ISSUE #2100: Need a way to obtain webchat.d.ts file
Need a way to get access to webchat.d.ts file.

I tried building the repo, but it generates multiple d.ts files, and consolidating them is painful. It's really helpful in writing typescript code. Please provide a way to get it, either directly from cdn or by building the code.

### PR DESCRIPTION
**Pull request by corinagum**
> Fixes #2100

## Changelog Entry

-  Resolves [#2100](https://github.com/microsoft/BotFramework-WebChat/issues/2100). Add types declarations for Style Options in api bundle, by [@corinagum](https://github.com/corinagum), in PR [#3818](https://github.com/microsoft/BotFramework-WebChat/pull/3818)

## Description

By popular request! This set of changes provides type declarations for Web Chat's style options.

## Design
Types file is exported to `/types` under `api` package.

## Specific Changes

- Clean up of StyleOption props
  - Several props were spread across two `defaultStyleOptions.js` :o
  - Update several props from `null` / empty string default value to `undefined`
  - Alphabetize props by category
- Delete `defaultStyleOptions.js` in `/component/Styles/` package
- Add typescript linting to eslint

-  [ ] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] I have updated documentation

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: __tests__/types/typesCheck.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const ts = require('typescript');
+const path = require('path');
+
+function compile(...filenames) {
+const program = ts.createProgram(filenames, {
```
Comment (compulim) [inline __tests__/types/typesCheck.js:5]: Indent.

File: __tests__/types/typesCheck.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+it('dir should pass dir as string', () => {
+  const dirStringErrors = compile(path.join(__dirname, './dirString.tsx'));
+
+  expect(dirStringErrors).toHaveProperty('length', 0);
+});
+
+it('dir should fail on dir as number', () => {
+  const dirNumErrors = compile(path.join(__dirname, './dirNumber.tsx'));
+
+  expect(dirNumErrors).toHaveProperty('length', 1);
```
Comment (compulim) [inline __tests__/types/typesCheck.js:41]: Can we assert part of the error message?

File: __tests__/types/typesCheck.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  }
+});
+
+if (errors.length > 0)
+
+return errors;
+};
+
+
+it('dir should pass dir as string', () => {
```
Comment (compulim) [inline __tests__/types/typesCheck.js:32]:
```suggestion
it('should pass dir as string', () => {
```

So it can read as a sentence "it should pass dir as string". This is called BDD.

File: packages/api/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// Please include any changes you make to this file to tsconfig.json files array
```
Comment (compulim) [inline packages/api/src/index.tsx:1]: We don't need this comment.

`tsconfig.json/files` lists all entrypoints and TSC will crawl thru the tree from these entrypoints.

File: packages/bundle/src/index.ts [hunk 5]
```diff
@@ -82,6 +84,8 @@ export {
   renderWebChat
 };
 
+export type { AdaptiveCardContent };
```
Comment (compulim) [inline packages/bundle/src/index.ts:87]: Do you recall why we need this line?

Comment (corinagum) [reply packages/bundle/src/index.ts:87]: no.. haha

File: packages/component/src/Styles/createStyleSet.ts [hunk 6]
```diff
@@ -30,7 +32,7 @@ import createSuggestedActionStyle from './StyleSet/SuggestedAction';
 import createTextContentStyle from './StyleSet/TextContent';
 import createToasterStyle from './StyleSet/Toaster';
 import createToastStyle from './StyleSet/Toast';
-import createTypingAnimationStyle from './StyleSet/TypingAnimation';
+import createTypingAnimationStyle from './StyleSet/TypingAnimation.js';
```
Comment (compulim) [inline packages/component/src/Styles/createStyleSet.ts:35]:
```suggestion
import createTypingAnimationStyle from './StyleSet/TypingAnimation';
```

File: packages/component/src/Styles/createStyleSet.ts [hunk 7]
```diff
@@ -39,14 +41,12 @@ import createVimeoContentStyle from './StyleSet/VimeoContent';
...
-import defaultStyleOptions from './defaultStyleOptions';
-
 // TODO: [P4] We should add a notice for people who want to use "styleSet" instead of "styleOptions".
 //       "styleSet" is actually CSS stylesheet and it is based on the DOM tree.
 //       DOM tree may change from time to time, thus, maintaining "styleSet" becomes a constant effort.
 
-export default function createStyleSet(options) {
-  options = { ...defaultStyleOptions, ...options };
+export default function createStyleSet(options: StyleOptions) {
+  options = { ...options };
```
Comment (compulim) [inline packages/component/src/Styles/createStyleSet.ts:49]: Removing patch code if they already patched in `api/patchStyleOptions.ts`.

File: packages/playground/src/App.js [hunk 8]
```diff
@@ -199,9 +199,9 @@ function App() {
   }, [handleUseMockBot]);
 
   // Feel free to change the text to whatever will help you dev faster
-  const handleCurrentCommandClick = useCallback(() => {
+  const handleCurrentCommandClick = () => {
```
Comment (compulim) [inline packages/playground/src/App.js:202]: This line seems unrelated to the PR.

**Review by compulim — APPROVED** [review]
Minor changes.

File: packages/playground/src/App.js (hunk 8 above)
Comment (corinagum) [reply packages/playground/src/App.js:202]: Yeah it's just a correction for playground

Comment (compulim) [inline packages/playground/src/App.js:202]:
```suggestion
  const handleCurrentCommandClick = useCallback(() => {
    store.dispatch({ type: 'WEB_CHAT/SEND_MESSAGE', payload: { text: 'suggested-actions' } });
  }, []);
```
